### DISCUSSION TITLE
Modal component joyride

### DISCUSSION THREAD

**Discussion by ofadevtr** [gilbarbara/react-joyride#998] [Q&A] [answered]
When running the joyride package. The 1st step is in the page, the 2nd step is in a modal. When the user says next to the 1st step, I want the modal to open and the 2nd step to be shown. how can I do it ?

**Comment by gilbarbara** [comment, marked as answer]
Hey @omerfarukaydins 

You'll probably need to use a controlled tour (using `stepIndex` that you handle manually) and use the callback to stop the tour, open your modal, update the stepIndex, and rerun the tour.
Check the **controlled** example in the demo.
